libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -170,24 +180,61 @@ impl Greenlight {
         seed: Vec<u8>,
         device: Device,
         persister: Arc<SqliteStorage>,
-    ) -> NodeResult<Greenlight> {
+    ) -> NodeResult<Self> {
         let greenlight_network = sdk_config.network.into();
-        let signer = Signer::new(seed, greenlight_network, device.clone())?;
-
+        let signer = Signer::new(seed.clone(), greenlight_network, device.clone())?;
+        let (tx, _) = mpsc::channel(1);
         Ok(Greenlight {
             sdk_config,
-            signer,
+            signer: Mutex::new(Arc::new(signer)),
             device,
+            seed,
             gl_client: Mutex::new(None),
             node_client: Mutex::new(None),
             persister,
             inprogress_payments: AtomicU16::new(0),
+            internal_shutdown: Mutex::new(Arc::new(tx)),
+            external_shutdown: Mutex::new(None),
         })
     }
 
+    async fn do_start_signer(&self) {
```

Comment:
Previously the caller has started the signer and was responsible for running it in a task so the greenlight module doesn't need to handle any async tasks. 
I think we can simplify by keeping it as it was and on reconnect only replace the internal signer, then the caller (who calls reconnect) will have to run the signer again (which we are doing already by restarting the background task).
As a result I think we also won't need also the internal shutdown lock at this context so everything related to the hibernation is solely coded on the `reconnect` function.

Reply:
This is now implemented, let me know what you think.